### PR TITLE
build: use faster minifier for js and css

### PR DESCRIPTION
**Pull request by anonrig**
Uses esbuild and lightningCSS to minify JS and CSS.

Running `yarn build-production` command:

Before:

```
webpack 5.87.0 compiled with 1 warning in 33069 ms
✨  Done in 34.68s.
```

After:

```
webpack 5.87.0 compiled with 16 warnings in 22640 ms
✨  Done in 24.21s.
```

### CODE REVIEW

File: webpack.config.ts [hunk 1]
```diff
@@ -338,14 +342,12 @@ const appConfig: Configuration = {
...
-        EXPERIMENTAL_SPA: JSON.stringify(SENTRY_EXPERIMENTAL_SPA),
-        SPA_DSN: JSON.stringify(SENTRY_SPA_DSN),
-        SENTRY_RELEASE_VERSION: JSON.stringify(SENTRY_RELEASE_VERSION),
-      },
+      'process.env.NODE_ENV': JSON.stringify(env.NODE_ENV),
+      'process.env.IS_ACCEPTANCE_TEST': JSON.stringify(IS_ACCEPTANCE_TEST),
+      'process.env.DEPLOY_PREVIEW_CONFIG': JSON.stringify(DEPLOY_PREVIEW_CONFIG),
+      'process.env.EXPERIMENTAL_SPA': JSON.stringify(SENTRY_EXPERIMENTAL_SPA),
+      'process.env.SPA_DSN': JSON.stringify(SENTRY_SPA_DSN),
+      'process.env.SENTRY_RELEASE_VERSION': JSON.stringify(SENTRY_RELEASE_VERSION),
```
Comment (evanpurkhiser) [inline webpack.config.ts:350]: Change looks unrelated?

File: webpack.config.ts [hunk 2]
```diff
@@ -470,14 +473,24 @@ const appConfig: Configuration = {
...
+        minify: TerserPlugin.esbuildMinify,
+      }),
+      new CssMinimizerPlugin({
+        parallel: true,
+        minify: CssMinimizerPlugin.lightningCssMinify,
+        minimizerOptions: {
+          targets: lightningcss.browserslistToTargets(
+            browserslist(packageJson.browserslist.production)
+          ),
+        } as never,
```
Comment (evanpurkhiser) [inline webpack.config.ts:491]: Why did you have to cast it?

File: yarn.lock [hunk 3]
```diff
@@ -4339,6 +4467,16 @@ browserslist@^4.0.0, browserslist@^4.14.5, browserslist@^4.16.6, browserslist@^4
     node-releases "^2.0.13"
     update-browserslist-db "^1.0.13"
 
+browserslist@^4.22.2:
```
Comment (evanpurkhiser) [inline yarn.lock:4470]: Can this get merged with the browserslist dep above

File: yarn.lock [hunk 4]
```diff
@@ -4461,6 +4599,11 @@ caniuse-lite@^1.0.0, caniuse-lite@^1.0.30001283, caniuse-lite@^1.0.30001541:
   resolved "https://registry.yarnpkg.com/caniuse-lite/-/caniuse-lite-1.0.30001549.tgz#7d1a3dce7ea78c06ed72c32c2743ea364b3615aa"
   integrity sha512-qRp48dPYSCYaP+KurZLhDYdVE+yEyht/3NlmcJgVQ2VMGt6JL36ndQ/7rgspdZsJuxDPFIo/OzBT2+GmIJ53BA==
 
+caniuse-lite@^1.0.30001565:
```
Comment (evanpurkhiser) [inline yarn.lock:4602]: Same here. should just upgrade the above version.

File: yarn.lock [hunk 5]
```diff
@@ -5436,6 +5584,11 @@ electron-to-chromium@^1.4.535:
   resolved "https://registry.yarnpkg.com/electron-to-chromium/-/electron-to-chromium-1.4.557.tgz#f3941b569c82b7bb909411855c6ff9bfe1507829"
   integrity sha512-6x0zsxyMXpnMJnHrondrD3SuAeKcwij9S+83j2qHAQPXbGTDDfgImzzwgGlzrIcXbHQ42tkG4qA6U860cImNhw==
 
+electron-to-chromium@^1.4.601:
```
Comment (evanpurkhiser) [inline yarn.lock:5587]: Also seems like it should merge with the one above.

I think you can use `yarn-deduplicate`

File: yarn.lock [hunk 6]
```diff
@@ -10972,6 +11232,16 @@ terser@5.16.9, terser@^5.10.0, terser@^5.16.5:
...
 
+terser@^5.16.8:
+  version "5.26.0"
+  resolved "https://registry.yarnpkg.com/terser/-/terser-5.26.0.tgz#ee9f05d929f4189a9c28a0feb889d96d50126fe1"
+  integrity sha512-dytTGoE2oHgbNV9nTzgBEPaqAWvcJNl66VZ0BkJqlvp71IjO8CxdBx/ykCNb47cLnCmCvRZ6ZR0tLkqvZCdVBQ==
+  dependencies:
+    "@jridgewell/source-map" "^0.3.3"
+    acorn "^8.8.2"
+    commander "^2.20.0"
+    source-map-support "~0.5.20"
```
Comment (scttcper) [inline yarn.lock:11243]: might want to do `npx yarn-deduplicate && yarn install` to reduce a few of these duplicates that should be able to be combined

Comment (scttcper) [reply yarn.lock:11243]: oh evan said the same thing

**Comment by evanpurkhiser** [comment]
Looking good. Think we just need to clean up the lock file!

File: webpack.config.ts (hunk 2 above)
Comment (anonrig) [reply webpack.config.ts:491]: `minimizerOptions.targets` is lighhtningCss specific, but I found a solution 👍

File: webpack.config.ts [hunk 7]
```diff
@@ -470,14 +479,24 @@ const appConfig: Configuration = {
       chunks: 'async',
       maxInitialRequests: 10, // (default: 30)
       maxAsyncRequests: 10, // (default: 30)
-      cacheGroups: {
-        ...localeChunkGroups,
-      },
+      cacheGroups,
```
Comment (JonasBa) [inline webpack.config.ts:482]: 👏🏼

File: webpack.config.ts [hunk 8]
```diff
@@ -449,6 +457,7 @@ const appConfig: Configuration = {
 
     modules: ['node_modules'],
     extensions: ['.jsx', '.js', '.json', '.ts', '.tsx', '.less'],
+    symlinks: false,
```
Comment (JonasBa) [inline webpack.config.ts:460]: q: Whats the reason for this change?

**Review by JonasBa — COMMENTED** [review]
Looks good 👏🏼 . As a final check, I would just compare the output and CSS size of the two minifiers? This seems like it is doing the work much faster, but I wonder if we are sacrificing some minification here?

File: webpack.config.ts (hunk 8 above)
Comment (anonrig) [reply webpack.config.ts:460]: Referencing Webpack documentation for improving the resolve speed: https://webpack.js.org/guides/build-performance/#resolving

```
Resolving
The following steps can increase resolving speed:

...
- Set resolve.symlinks: false if you don't use symlinks (e.g. npm link or yarn link).
...
```

File: webpack.config.ts (hunk 7 above)
Comment (evanpurkhiser) [reply webpack.config.ts:482]: I was doing this for documentation purposes. Changing the name of this variable loses the information that it was maintaining the lits of chunk groups used for locale-based chunking

Comment (anonrig) [reply webpack.config.ts:482]: The previous code was making a copy of the object. `cacheGroups: localeChunkGroups` could be the correct approach, but I'd argue for why not include a comment before the variable declaration but indeed I'd go for non-copying solution.

Comment (evanpurkhiser) [reply webpack.config.ts:482]: Yeah, I think the documentation purposes this served outweighed whatever negligible memory benefits we're gaining here.

**Comment by ryan953** [comment]
in the description there is the Before speed and After speed. Can you write down the command that was executed so archaeologists know what to run.

**Comment by anonrig** [comment]
> in the description there is the Before speed and After speed. Can you write down the command that was executed so archaeologists know what to run.

Makes sense. Updated.

**Comment by anonrig** [comment]
@getsentry/app-frontend Appreciate a review on this.

**Review by scttcper — APPROVED** [review]
looked good when i tried it

**Review by malwilley — APPROVED** [review]
Tested all the stack trace components which I know use some of the  .less styles. Looks good to me.

**Comment by anonrig** [comment]
I rebased and force-pushed. Hopefully, we can land this today.

**Review by scttcper — APPROVED** [review]
